escrow/SPECIFICATION.md

```diff
@@ -0,0 +1,95 @@
+Table of Content
+- [Actions that users are able to perform](#actions-that-users-are-able-to-perform)
+  - [Arbiter](#arbiter)
+    - [`resolve_dispute()`](#resolve_dispute)
+  - [Buyer](#buyer)
+    - [`accept_arbiter()`](#accept_arbiter)
+    - [`deposit()`](#deposit)
+    - [`dispute()`](#dispute)
+    - [`transfer_to_seller()`](#transfer_to_seller)
+  - [Seller](#seller)
+    - [`create_escrow()`](#create_escrow)
+    - [`propose_arbiter()`](#propose_arbiter)
+    - [`return_deposit()`](#return_deposit)
+    - [`take_payment()`](#take_payment)
+
+# Overview
+
+This document provides an overview of the application.
+
+It outline the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
+
+# Use Cases
+
+This section contains general information about the functionality of the application and thus does not touch upon any technical aspects.
+
+If you are interested in a functional overview then this is the section for you.
+
+## Actions that users are able to perform
+
+This sub-section details what a user is able to do e.g. click a button and "x, y, z" happens.
+
+### Arbiter
+
+#### `resolve_dispute()`
+
+1. When an escrow is in dispute the arbiter of the escrow can perform a resolution in either direction of the buyer or seller
+2. A resolution consists of
+   1. Selecting either the buyer or seller to whom the deposited asset from the buyer will be sent to
+   2. Selecting an amount to take as payment from the collateral provided by the seller
+
+### Buyer
+
+#### `accept_arbiter()`
+
+1. The buyer is able to accept a proposition by the seller to change the arbiter or change the fee for the arbiter
+
+> **NOTE** They are not required to do so however there may be instances where changing the arbiter / fee is favourable
+
+#### `deposit()`
+
+1. The buyer must deposit into the escrow one asset from the options provided by the seller
+2. The escrow accepts a 1 time deposit therefore the buyer cannot go back and deposit a different asset
```

Comment:
```suggestion
2. The escrow accepts a 1 time deposit.  Therefore the buyer cannot go back and deposit a different asset
```

Reply:
This is also incorrect

Reply:
1. hmm I'm not sure this is incorrect.  I think the "therefore" needs some sort of punctuation around it since you are using it to connect two [independent clauses](https://www.wikihow.com/Use-Therefore-in-a-Sentence), but also grammar is not my strongest lol.

Reply:
Also I think the original sentence works well enough if you don't feel like changin.

Reply:
Same here. Looking at that link I don't know why they end the sentence and start with "Therefore". I'm pretty sure that is incorrect or rather it can be done but that's not the correct way to use it. Afaik it's meant to connect and the common use may / must involve a comma at times

Reply:
Feel free to add punctuation or don't.  I think readers will be able to understand the original.

Reply:
Deep into bike-shedding territory here, but technically, it's either:

> "The escrow accepts a one-time deposit.  Therefore, the buyer ... " 

or

> "The escrow accepts a one-time deposit; therefore, the buyer ... "

Reply:
hm, I thought something like the second one was the correct way to do it while the first is incorrect. Guess I was never taught the first way